### DISCUSSION TITLE
Equivalent of Maps.snapshot() for MapsGrid??

### DISCUSSION THREAD

**Discussion by Joshdpaul** [raphaelquast/EOmaps#204] [Q&A]
I am looking for a way to render a static MapsGrid object in the same manner as the `Maps.snapshot()` [function](https://eomaps.readthedocs.io/en/latest/generated/eomaps.eomaps.Maps.snapshot.html). This is when using `eomaps` inside of Jupyter Lab. The intention is to be able to share the notebook map grid graphics when the notebook is rendered in Github, rendered as HTML, exported as PDF, etc.

Any suggestions?  Thanks in advance!

**Comment by raphaelquast** [comment]
Hey, you are right...  a `MapsGrid` should actually get it's own `.snapshot` and `.show` methods... 
> I'll try to get this into EOmaps in upcoming releases.... I'm also happy to accept any Pull Request that addresses this!


For now, there are 2 possibilities:

### If you want to keep using MapsGrid:
You can acutally call `.show` from any `Maps` object that belongs to the same figure... using `mg.parent` will select the first Maps-object
```python
from eomaps import MapsGrid

mg = MapsGrid(2, 2, figsize=(8, 4.5))
mg.add_feature.preset.coastline()
mg.parent.show()   # just use one of the Maps objects of the grid to call .show() or .snapshot()
```

### If you are OK with creating the individual maps yourself:

(see the [Multiple maps and/or plots in one figure](https://eomaps.readthedocs.io/en/latest/api_basics.html#multiple-maps-and-or-plots-in-one-figure) section for details on what the `ax` kwarg means)

```python
from eomaps import Maps
m = Maps(ax=221, figsize=(8, 4.5))
all_maps = [m]
for ax in [222, 223, 224]:
    all_maps.append(m.new_map(ax=ax))

for m in all_maps:
    m.add_feature.preset("coastline", "ocean", "land")

m.show()
```

**Comment by raphaelquast** [comment]
Starting with [EOmaps v7.3.3](https://github.com/raphaelquast/EOmaps/releases/tag/v7.3.3), `MapsGrid` objects now have dedicated `.show()` and `.snapshot()` methods ! 

Thanks @Joshdpaul for taking the time to report this!
